docs/recipes/endpoint-testing-with-mongoose.md

```diff
@@ -0,0 +1,281 @@
+# Endpoint testing with Mongoose
+
+This recipe shows you how to test your endpoints with AVA and Mongoose.
+
+## Setup
+
+This recipe uses the following libraries:
+
+1. [MongoDB memory server](https://github.com/nodkz/mongodb-memory-server) (A MongoDB in-memory Server)
+2. [Babel Polyfill](https://babeljs.io/docs/usage/polyfill/) (required for MongoDB memory server)
+3. [Supertest](https://github.com/visionmedia/supertest) (An endpoint testing library)
+4. Mongoose
+
+Install the first three libraries by running the following code:
+
+```console
+$ npm install --save-dev mongodb-memory-server babel-polyfill supertest
+```
+
+You should have Mongoose installed already. If not, run the following code to install it:
+
+(Note: You need at least mongoose v4.11.3)
+
+```console
+$ npm install mongoose
+```
+
+## Setting up AVA
+
+Since MongoDB Memory server requires Babel polyfill to work, the easiest way to set up AVA is through the `ava` key in your `package.json` file.
+
+```json
+"ava": {
+    "require": [
+      "babel-register",
+      "babel-polyfill"
+    ]
+  },
+```
+
+If you want AVA to run your tests whenever changes are made, add the `--watch` option.
+
+```json
+"scripts": {
+  "watch": "ava --watch"
+},
+```
+
+## Your test file
+
+First, include the libraries you need.
+
+```js
+// Libraries required for testing
+import test from 'ava'
+import request from 'supertest'
+import MongodbMemoryServer from 'mongodb-memory-server'
+import mongoose from 'mongoose'
+
+// Your server and models
+import app from '../server'
+import User from '../models/User'
+```
+
+### Your server file
+
+In your `server` file, you cannot use `app.listen` to start your app. Export it to another file and call `app.listen` there.
+
+This `server` file should resemble the following:
```

Comment:
I'm thinking this should just link to https://github.com/zellwk/ava-mdb-test/blob/master/server.js. Not sure it it needs to be copied into the test. Similarly you may want to link to https://github.com/zellwk/ava-mdb-test/blob/master/models/User.js which, not knowing anything about Mongoose, I find quite relevant.